openhcl/virt_mshv_vtl/src/processor/tdx/mod.rs

```diff
@@ -2023,6 +2062,93 @@ impl UhProcessor<'_, TdxBacked> {
+                tracing::error!(
+                    ?cached_processor_controls,
+                    ?cached_secondary_processor_controls,
+                    vmcs_processor_controls,
+                    vmcs_secondary_processor_controls,
+                    "processor controls"
+                );
+
+                let cached_tpr_threshold = self.backing.vtls[vtl].tpr_threshold;
+                let vmcs_tpr_threshold = self
+                    .runner
+                    .read_vmcs32(vtl, VmcsField::VMX_VMCS_TPR_THRESHOLD);
+                tracing::error!(cached_tpr_threshold, vmcs_tpr_threshold, "tpr threshold");
+
+                let cached_eoi_exit_bitmap = self.backing.eoi_exit_bitmap;
+                let vmcs_eoi_exit_bitmap = {
+                    let fields = [
+                        VmcsField::VMX_VMCS_EOI_EXIT_0,
+                        VmcsField::VMX_VMCS_EOI_EXIT_1,
+                        VmcsField::VMX_VMCS_EOI_EXIT_2,
+                        VmcsField::VMX_VMCS_EOI_EXIT_3,
+                    ];
+                    fields
+                        .iter()
+                        .map(|field| self.runner.read_vmcs32(vtl, *field))
+                        .collect::<Vec<_>>()
+                };
+                tracing::error!(
+                    ?cached_eoi_exit_bitmap,
+                    ?vmcs_eoi_exit_bitmap,
+                    "eoi exit bitmap"
+                );
+
+                let cached_interrupt_information = self.backing.vtls[vtl].interruption_information;
+                let cached_interruption_set = self.backing.vtls[vtl].interruption_set;
+                let vmcs_interrupt_information = self
+                    .runner
+                    .read_vmcs32(vtl, VmcsField::VMX_VMCS_ENTRY_INTERRUPT_INFO);
+                let vmcs_entry_exception_code = self
+                    .runner
+                    .read_vmcs32(vtl, VmcsField::VMX_VMCS_ENTRY_EXCEPTION_ERROR_CODE);
+                tracing::error!(
+                    ?cached_interrupt_information,
+                    cached_interruption_set,
+                    vmcs_interrupt_information,
+                    vmcs_entry_exception_code,
+                    "interrupt information"
+                );
+
+                let guest_interruptibility = self
+                    .runner
+                    .read_vmcs32(vtl, VmcsField::VMX_VMCS_GUEST_INTERRUPTIBILITY);
+                tracing::error!(guest_interruptibility, "guest interruptibility");
+
+                let vmcs_sysenter_cs = self
+                    .runner
+                    .read_vmcs32(vtl, VmcsField::VMX_VMCS_GUEST_SYSENTER_CS_MSR);
+                let vmcs_sysenter_esp = self
+                    .runner
+                    .read_vmcs32(vtl, VmcsField::VMX_VMCS_GUEST_SYSENTER_ESP_MSR);
+                let vmcs_sysenter_eip = self
+                    .runner
+                    .read_vmcs32(vtl, VmcsField::VMX_VMCS_GUEST_SYSENTER_EIP_MSR);
```

Comment:
This must be vmcs64 or this will panic.

I'd suggest doing something locally to force this code to run to make sure there are no other such bugs.

Reply:
yeah i moved this logging into its own fn so it can be tested. I'm going to submit a follow up that changes how we do vmcs acceses so you can't misuse it anymore (there was another one that panicked too)

Reply:
#1118